WordPressKit/FeatureFlagRemote.swift

```diff
@@ -60,7 +55,7 @@ open class FeatureFlagRemote: ServiceRemoteWordPressComREST {
                                 })
     }
 
-    public struct RemoteFeatureFlagsEndpointParams {
+    public struct FetchAllEndpointParams {
```

Comment:
@salimbraksa `FetchAllEndpointParams` feels a bit confusing. How about `SessionDetails`?

Also, can we move it outside of the `FeatureFlagRemote` scope? Referencing it by `FeatureFlagRemote.FetchAllEndpointParams` hurts the readability as it implies it is tighly coupled with feature flags while it's not. This is what I meant here https://github.com/wordpress-mobile/WordPressKit-iOS/pull/674#discussion_r1442456627

Reply:
My concern with global scope is the following:
1. Let's say another developer added a valid "session" attribute in `SessionDetails`. 
2. The new attribute is encoded and returned in the dictionary representation.
3. Now that attribute will be sent in all other endpoints.

This is why I kept that type tightly related to `Feature Flags`, and if you think about it, the new params added in Dashboard Cards are for the Feature Flags mechanism.

Reply:
> Now that attribute will be sent in all other endpoints, and I'm not sure how endpoints handle extra attribute that don't need.

We can assume that developers will seek to understand what this type is used for before changing it.
However, extra params won't cause an issue either way.

Reply:
> We can assume that developers will seek to understand what this type is used for before changing it.

I agree, but my concern is that if we expand the SessionDetails with new 'session' attributes in the future, we need to ensure they are not inadvertently included in the Feature Flags and Dashboard Cards endpoint query params.

If we rename that type to `SessionDetails`, I think the Feature Flag Remote and Dashboard Cards service should select only the params they need.

Also, while I'm sure developers double check API usage before making changes, I think it's also important to write resilient code that mitigates the risks associated with human error.

Reply:
I'll rename the type to `SessionsDetails` and update the Unit Tests to ensure **only** the the params required by the endpoint are sent.

Reply:
> However, extra params won't cause an issue either way.

Given this, I think we would be over-engineering to avoid something that's not critical.

Reply:
Resolved in 28cc006f19a92ddb7e0b307926cedd1c9449bca4

> However, extra params won't cause an issue either way.

Passing extra params to endpoints is a code smell in my opinion. Moreover, some backend systems may reject additional parameters as a security measure.

I agree that we shouldn't spend time on imaginary problems. For now, it's enough to have Unit Tests to check for this. I looked at the tests we already have, and they'll fail if extra params are sent to Feature Flags and Dashboard Cards endpoints.